pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -163,28 +163,13 @@ stdenv.mkDerivation {
     ./0009-Change-usr-share-zoneinfo-to-etc-zoneinfo.patch
     ./0010-localectl-use-etc-X11-xkb-for-list-x11.patch
     ./0011-build-don-t-create-statedir-and-don-t-touch-prefixdi.patch
-    ./0012-inherit-systemd-environment-when-calling-generators.patch
-    ./0013-add-rootprefix-to-lookup-dir-paths.patch
-    ./0014-systemd-shutdown-execute-scripts-in-etc-systemd-syst.patch
-    ./0015-systemd-sleep-execute-scripts-in-etc-systemd-system-.patch
-    ./0016-kmod-static-nodes.service-Update-ConditionFileNotEmp.patch
-    ./0017-path-util.h-add-placeholder-for-DEFAULT_PATH_NORMAL.patch
-    ./0018-pkg-config-derive-prefix-from-prefix.patch
-
-    # In v248 or v249 we started to get in trouble due to our
-    # /etc/systemd/system being a symlink and thus being treated differently by
-    # systemd. With the below patch we mitigate that effect by special casing
-    # all our root unit dirs if they are symlinks. This does exactly what we
-    # need (AFAICT).
-    # See https://github.com/systemd/systemd/pull/20479 for upstream discussion.
-    ./0019-core-handle-lookup-paths-being-symlinks.patch
-
-    # fixes reproducability of dbus xml files
-    # Should no longer be necessary with v251.
-    (fetchpatch {
-      url = "https://github.com/systemd/systemd/pull/22174.patch";
-      sha256 = "sha256-RVhxUEUiISgRlIP/AhU+w1VHfDQw2W16cFl2TXXyxno=";
-    })
+    ./0012-add-rootprefix-to-lookup-dir-paths.patch
+    ./0013-systemd-shutdown-execute-scripts-in-etc-systemd-syst.patch
+    ./0014-systemd-sleep-execute-scripts-in-etc-systemd-system-.patch
+    ./0015-kmod-static-nodes.service-Update-ConditionFileNotEmp.patch
+    ./0016-path-util.h-add-placeholder-for-DEFAULT_PATH_NORMAL.patch
+    ./0017-pkg-config-derive-prefix-from-prefix.patch
+    ./0018-inherit-systemd-environment-when-calling-generators.patch
   ] ++ lib.optional stdenv.hostPlatform.isMusl (
     let
       oe-core = fetchzip {
```

Comment:
@alyssais are there already musl patches for this version of systemd?

Reply:
[Yes](https://git.openembedded.org/openembedded-core/commit/meta/recipes-core/systemd?id=3275cc6c2ecfd70ba134b49363c9cecbb0c2c0b2)

Reply:
Usually I'd plan on updating the Musl patches myself after the main systemd PR was merged (to avoid friction, since most of our systemd maintainers don't want to have to think about Musl), but you could also incorporate them into this PR if you'd prefer.

Reply:
Happy with either approach. @Mic92 i'm happy to merge as is or wait until you've added the patches

Reply:
@alyssais: it's in master, and musl is failing.  It doesn't suffice to simply drop the subset that doesn't apply.

Reply:
Whoops, I missed that this was merged.  I'll get that fixed.

Reply:
Cross-ref: https://github.com/NixOS/nixpkgs/pull/183721